Pyaerocom ValueError when running with numpy 2.0.0

**Describe the bug**
Please provide a clear and concise description of what the bug is.
 - Pyaerocom version: 0.20.dev1
 - Computing platform: PPI
 - Configuration file (if applicable): NA
 - Error message (if applicable): NA

**To Reproduce**
Steps to reproduce the behavior:
1. `pip install numpy==2.0.0`
2. `pytest .`

**Expected behavior**
No error should occur

**Screenshots**
```
ImportError while loading conftest '/lustre/storeB/users/thlun8736/python/pyaerocom/tests/conftest.py'.
tests/conftest.py:6: in <module>
    from pyaerocom import const
pyaerocom/__init__.py:9: in <module>
    from .config import Config
pyaerocom/config.py:19: in <module>
    from pyaerocom.grid_io import GridIO
pyaerocom/grid_io.py:2: in <module>
    from pyaerocom.time_config import TS_TYPES
pyaerocom/time_config.py:7: in <module>
    from iris import coord_categorisation
venv/lib/python3.11/site-packages/iris/coord_categorisation.py:23: in <module>
    import iris.coords
venv/lib/python3.11/site-packages/iris/coords.py:23: in <module>
    from iris.common import (
venv/lib/python3.11/site-packages/iris/common/__init__.py:9: in <module>
    from .mixin import *
venv/lib/python3.11/site-packages/iris/common/mixin.py:10: in <module>
    import cf_units
venv/lib/python3.11/site-packages/cf_units/__init__.py:23: in <module>
    from cf_units import _udunits2 as _ud
cf_units/_udunits2.pyx:1: in init cf_units._udunits2
    ???
E   ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

**Additional context**
- https://github.com/SciTools/iris/issues/6043

--- a/pyaerocom/extras/satellite_l2/aeolus_l2a.py
+++ b/pyaerocom/extras/satellite_l2/aeolus_l2a.py
@@ -715,7 +715,7 @@ class ReadL2Data(ReadL2DataBase):
         seconds_to_add = np.datetime64("2000-01-01T00:00:00") - np.datetime64(
             "1970-01-01T00:00:00"
         )
-        seconds_to_add = seconds_to_add.astype(np.float_)
+        seconds_to_add = seconds_to_add.astype(np.float64)
 
         # the same can be achieved using pandas, but we stick to numpy here
         # base_time = pd.DatetimeIndex(['2000-01-01'])
@@ -832,9 +832,9 @@ class ReadL2Data(ReadL2DataBase):
                 # return as one multidimensional numpy array that can be put into self.data directly
                 # (column wise because the column numbers do not match)
                 index_pointer = 0
-                data = np.empty([self._ROWNO, self._COLNO], dtype=np.float_)
+                data = np.empty([self._ROWNO, self._COLNO], dtype=np.float64)
 
-                for idx, _time in enumerate(file_data["time"].astype(np.float_)):
+                for idx, _time in enumerate(file_data["time"].astype(np.float64)):
                     # skip times of profiles without a single valid extinction
                     # the following is deprecated in current nympy
                     # if _time in times_to_skip:
@@ -912,7 +912,7 @@ class ReadL2Data(ReadL2DataBase):
 
                         if index_pointer >= self._ROWNO:
                             # add another array chunk to self.data
-                            chunk = np.empty([self._CHUNKSIZE, self._COLNO], dtype=np.float_)
+                            chunk = np.empty([self._CHUNKSIZE, self._COLNO], dtype=np.float64)
                             data = np.append(data, chunk, axis=0)
 
                 # return only the needed elements...
@@ -1137,7 +1137,7 @@ class ReadL2Data(ReadL2DataBase):
         start = time.perf_counter()
 
         data = ungridded_data_obj._data
-        ret_data = np.empty([self._ROWNO, self._COLNO], dtype=np.float_)
+        ret_data = np.empty([self._ROWNO, self._COLNO], dtype=np.float64)
         index_counter = 0
         cut_flag = True
         matching_indexes = []
@@ -1164,7 +1164,7 @@ class ReadL2Data(ReadL2DataBase):
                                 ret_data,
                                 np.zeros(
                                     [end_index - len(ret_data), self._COLNO],
-                                    dtype=np.float_,
+                                    dtype=np.float64,
                                     axis=0,
                                 ),
                             )
@@ -1874,7 +1874,7 @@ class ReadL2Data(ReadL2DataBase):
         seconds_to_add = np.datetime64("2000-01-01T00:00:00") - np.datetime64(
             "1970-01-01T00:00:00"
         )
-        seconds_to_add = seconds_to_add.astype(np.float_)
+        seconds_to_add = seconds_to_add.astype(np.float64)
 
         # the same can be achieved using pandas, but we stick to numpy here
         # base_time = pd.DatetimeIndex(['2000-01-01'])
@@ -1952,7 +1952,7 @@ class ReadL2Data(ReadL2DataBase):
                         out_struct[codaRec._registeredFields[idx]] = {}
                         for str_name in dummy:
                             out_struct[codaRec._registeredFields[idx]][str_name] = np.empty(
-                                rec_length, dtype=np.float_
+                                rec_length, dtype=np.float64
                             )
 
                     for str_name in dummy:

--- a/pyaerocom/extras/satellite_l2/base_reader.py
+++ b/pyaerocom/extras/satellite_l2/base_reader.py
@@ -389,7 +389,7 @@ class ReadL2DataBase(ReadUngriddedBase):
                         index_store[_key] = file_data[_key].shape[0]
                         input_shape = list(file_data[_key].shape)
                         input_shape[0] = self._ROWNO
-                        data_obj._data[_key] = np.empty(input_shape, dtype=np.float_)
+                        data_obj._data[_key] = np.empty(input_shape, dtype=np.float64)
                         if len(input_shape) == 1:
                             data_obj._data[_key][0 : file_data[_key].shape[0]] = file_data[_key]
                         elif len(input_shape) == 2:
@@ -423,7 +423,7 @@ class ReadL2DataBase(ReadUngriddedBase):
                         if index_store[_key] + elements_to_add > data_obj._data[_key].shape[0]:
                             current_shape = list(data_obj._data[_key].shape)
                             current_shape[0] = current_shape[0] + self._CHUNKSIZE
-                            tmp_data = np.empty(current_shape, dtype=np.float_)
+                            tmp_data = np.empty(current_shape, dtype=np.float64)
                             if len(current_shape) == 1:
                                 tmp_data[0 : data_obj._data[_key].shape[0]] = data_obj._data[_key]
                             elif len(current_shape) == 2:

--- a/pyaerocom/extras/satellite_l2/sentinel5p.py
+++ b/pyaerocom/extras/satellite_l2/sentinel5p.py
@@ -124,7 +124,7 @@ class ReadL2Data(ReadL2DataBase):
         self.NAN_DICT.update({self._ALTITUDENAME: -1.0})
 
         # scaling factors e.g. for unit conversion
-        self.SCALING_FACTORS[self._NO2NAME] = np.float_(6.022140857e19 / 1.0e15)
+        self.SCALING_FACTORS[self._NO2NAME] = np.float64(6.022140857e19 / 1.0e15)
 
         # the following defines necessary quality flags for a value to make it into the used data set
         # the flag needs to have a HIGHER or EQUAL value than the one listed here
@@ -137,11 +137,11 @@ class ReadL2Data(ReadL2DataBase):
         self.CODA_READ_PARAMETERS[self._NO2NAME] = {}
         self.CODA_READ_PARAMETERS[self._NO2NAME]["metadata"] = {}
         self.CODA_READ_PARAMETERS[self._NO2NAME]["vars"] = {}
-        self.CODA_READ_PARAMETERS[self._NO2NAME]["time_offset"] = np.float_(24.0 * 60.0 * 60.0)
+        self.CODA_READ_PARAMETERS[self._NO2NAME]["time_offset"] = np.float64(24.0 * 60.0 * 60.0)
         self.CODA_READ_PARAMETERS[self._O3NAME] = {}
         self.CODA_READ_PARAMETERS[self._O3NAME]["metadata"] = {}
         self.CODA_READ_PARAMETERS[self._O3NAME]["vars"] = {}
-        self.CODA_READ_PARAMETERS[self._O3NAME]["time_offset"] = np.float_(24.0 * 60.0 * 60.0)
+        self.CODA_READ_PARAMETERS[self._O3NAME]["time_offset"] = np.float64(24.0 * 60.0 * 60.0)
 
         # self.CODA_READ_PARAMETERS[DATASET_NAME]['metadata'][_TIME_NAME] = 'PRODUCT/time_utc'
         self.CODA_READ_PARAMETERS[self._NO2NAME]["metadata"][self._TIME_NAME] = "PRODUCT/time"
@@ -293,7 +293,7 @@ class ReadL2Data(ReadL2DataBase):
             self.CODA_READ_PARAMETERS[self._AVERAGINGKERNELNAME] = {}
             self.CODA_READ_PARAMETERS[self._AVERAGINGKERNELNAME]["metadata"] = {}
             self.CODA_READ_PARAMETERS[self._AVERAGINGKERNELNAME]["vars"] = {}
-            self.CODA_READ_PARAMETERS[self._AVERAGINGKERNELNAME]["time_offset"] = np.float_(
+            self.CODA_READ_PARAMETERS[self._AVERAGINGKERNELNAME]["time_offset"] = np.float64(
                 24.0 * 60.0 * 60.0
             )
             self.CODA_READ_PARAMETERS[self._AVERAGINGKERNELNAME]["metadata"][self._TIME_NAME] = (
@@ -330,7 +330,7 @@ class ReadL2Data(ReadL2DataBase):
             self.CODA_READ_PARAMETERS[self._LEVELSNAME] = {}
             self.CODA_READ_PARAMETERS[self._LEVELSNAME]["metadata"] = {}
             self.CODA_READ_PARAMETERS[self._LEVELSNAME]["vars"] = {}
-            self.CODA_READ_PARAMETERS[self._LEVELSNAME]["time_offset"] = np.float_(
+            self.CODA_READ_PARAMETERS[self._LEVELSNAME]["time_offset"] = np.float64(
                 24.0 * 60.0 * 60.0
             )
             self.CODA_READ_PARAMETERS[self._LEVELSNAME]["vars"][self._LEVELSNAME] = "PRODUCT/layer"
@@ -338,7 +338,7 @@ class ReadL2Data(ReadL2DataBase):
             self.CODA_READ_PARAMETERS[self._GROUNDPRESSURENAME] = {}
             self.CODA_READ_PARAMETERS[self._GROUNDPRESSURENAME]["metadata"] = {}
             self.CODA_READ_PARAMETERS[self._GROUNDPRESSURENAME]["vars"] = {}
-            self.CODA_READ_PARAMETERS[self._GROUNDPRESSURENAME]["time_offset"] = np.float_(
+            self.CODA_READ_PARAMETERS[self._GROUNDPRESSURENAME]["time_offset"] = np.float64(
                 24.0 * 60.0 * 60.0
             )
             self.CODA_READ_PARAMETERS[self._GROUNDPRESSURENAME]["vars"][
@@ -349,7 +349,7 @@ class ReadL2Data(ReadL2DataBase):
             self.CODA_READ_PARAMETERS[self._TM5_TROPOPAUSE_LAYER_INDEX_NAME]["metadata"] = {}
             self.CODA_READ_PARAMETERS[self._TM5_TROPOPAUSE_LAYER_INDEX_NAME]["vars"] = {}
             self.CODA_READ_PARAMETERS[self._TM5_TROPOPAUSE_LAYER_INDEX_NAME]["time_offset"] = (
-                np.float_(24.0 * 60.0 * 60.0)
+                np.float64(24.0 * 60.0 * 60.0)
             )
             self.CODA_READ_PARAMETERS[self._TM5_TROPOPAUSE_LAYER_INDEX_NAME]["vars"][
                 self._TM5_TROPOPAUSE_LAYER_INDEX_NAME
@@ -358,7 +358,7 @@ class ReadL2Data(ReadL2DataBase):
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_A_NAME] = {}
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_A_NAME]["metadata"] = {}
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_A_NAME]["vars"] = {}
-            self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_A_NAME]["time_offset"] = np.float_(
+            self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_A_NAME]["time_offset"] = np.float64(
                 24.0 * 60.0 * 60.0
             )
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_A_NAME]["vars"][
@@ -367,7 +367,7 @@ class ReadL2Data(ReadL2DataBase):
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_B_NAME] = {}
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_B_NAME]["metadata"] = {}
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_B_NAME]["vars"] = {}
-            self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_B_NAME]["time_offset"] = np.float_(
+            self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_B_NAME]["time_offset"] = np.float64(
                 24.0 * 60.0 * 60.0
             )
             self.CODA_READ_PARAMETERS[self._TM5_CONSTANT_B_NAME]["vars"][
@@ -514,7 +514,7 @@ class ReadL2Data(ReadL2DataBase):
         seconds_to_add = np.datetime64("2010-01-01T00:00:00") - np.datetime64(
             "1970-01-01T00:00:00"
         )
-        seconds_to_add = seconds_to_add.astype(np.float_)
+        seconds_to_add = seconds_to_add.astype(np.float64)
 
         # the same can be achieved using pandas, but we stick to numpy here
         # base_time = pd.DatetimeIndex(['2000-01-01'])
@@ -591,7 +591,7 @@ class ReadL2Data(ReadL2DataBase):
             # return as one multidimensional numpy array that can be put into self.data directly
             # (column wise because the column numbers do not match)
             index_pointer = 0
-            data = np.empty([self._ROWNO, colno], dtype=np.float_)
+            data = np.empty([self._ROWNO, colno], dtype=np.float64)
             # loop over the times
             for idx, _time in enumerate(file_data[self._TIME_OFFSET_NAME]):
                 # loop over the number of ground pixels
@@ -610,9 +610,9 @@ class ReadL2Data(ReadL2DataBase):
 
                     # time can be a scalar...
                     try:
-                        data[index_pointer, self._TIMEINDEX] = _time.astype(np.float_)
+                        data[index_pointer, self._TIMEINDEX] = _time.astype(np.float64)
                     except Exception:
-                        data[index_pointer, self._TIMEINDEX] = _time[_index].astype(np.float_)
+                        data[index_pointer, self._TIMEINDEX] = _time[_index].astype(np.float64)
 
                     # loop over the variables
                     for var in vars_to_read_in:
@@ -633,7 +633,9 @@ class ReadL2Data(ReadL2DataBase):
                     if index_pointer >= self._ROWNO:
                         # add another array chunk to self.data
                         data = np.append(
-                            data, np.empty([self._CHUNKSIZE, self._COLNO], dtype=np.float_), axis=0
+                            data,
+                            np.empty([self._CHUNKSIZE, self._COLNO], dtype=np.float64),
+                            axis=0,
                         )
                         # unneeded after update (_ROWNO is now dynamic and returns shape index 0 of numpy array)
                         # self._ROWNO += self._CHUNKSIZE

--- a/pyaerocom/io/gaw/reader.py
+++ b/pyaerocom/io/gaw/reader.py
@@ -136,7 +136,7 @@ class ReadGAW(ReadUngriddedBase):
             if np.shape(data[i])[0] != 10:
                 del data[i]
 
-        data = np.array(data)
+        data = np.array(data, dtype=object)
 
         # names of the columns in the file that I want to use
         file_vars = file_vars[5:9]

--- a/pyaerocom/io/read_aeronet_invv3.py
+++ b/pyaerocom/io/read_aeronet_invv3.py
@@ -196,7 +196,7 @@ class ReadAeronetInvV3(ReadAeronetBase):
                 # copy the data fields that are available (rest will be filled
                 # below)
                 for var, idx in vars_available.items():
-                    val = np.float_(dummy_arr[idx])
+                    val = np.float64(dummy_arr[idx])
                     if val == self.NAN_VAL:
                         val = np.nan
                     data_out[var].append(val)

--- a/pyaerocom/io/read_aeronet_sdav3.py
+++ b/pyaerocom/io/read_aeronet_sdav3.py
@@ -199,7 +199,7 @@ class ReadAeronetSdaV3(ReadAeronetBase):
 
                 # copy the data fields
                 for var, idx in vars_available.items():
-                    val = np.float_(dummy_arr[idx])
+                    val = np.float64(dummy_arr[idx])
                     if val == self.NAN_VAL:
                         val = np.nan
                     data_out[var].append(val)

--- a/pyaerocom/io/read_aeronet_sunv3.py
+++ b/pyaerocom/io/read_aeronet_sunv3.py
@@ -266,7 +266,7 @@ class ReadAeronetSunV3(ReadAeronetBase):
             data_out["dtime"].append(np.datetime64(datestring))
 
             for var, idx in vars_available.items():
-                val = np.float_(dummy_arr[idx])
+                val = np.float64(dummy_arr[idx])
                 if val == self.NAN_VAL:
                     val = np.nan
                 data_out[var].append(val)

--- a/pyaerocom/io/read_eea_aqerep_base.py
+++ b/pyaerocom/io/read_eea_aqerep_base.py
@@ -112,16 +112,16 @@ class ReadEEAAQEREPBase(ReadUngriddedBase):
     # conversion factor between concX and vmrX
     CONV_FACTOR = {}
 
-    CONV_FACTOR["concSso2"] = np.float_(0.50052292274792)
-    CONV_FACTOR["concNno2"] = np.float_(0.3044517868011477)
-    CONV_FACTOR["concNno"] = np.float_(0.466788868521913)
-    CONV_FACTOR["vmro3"] = np.float_(
+    CONV_FACTOR["concSso2"] = np.float64(0.50052292274792)
+    CONV_FACTOR["concNno2"] = np.float64(0.3044517868011477)
+    CONV_FACTOR["concNno"] = np.float64(0.466788868521913)
+    CONV_FACTOR["vmro3"] = np.float64(
         0.493
     )  # retrieved using STD atmosphere from geonum and pya.mathutils.concx_to_vmrx
-    CONV_FACTOR["vmro3max"] = np.float_(
+    CONV_FACTOR["vmro3max"] = np.float64(
         0.493
     )  # retrieved using STD atmosphere from geonum and pya.mathutils.concx_to_vmrx
-    CONV_FACTOR["vmrno2"] = np.float_(
+    CONV_FACTOR["vmrno2"] = np.float64(
         0.514
     )  # retrieved using STD atmosphere from geonum and pya.mathutils.concx_to_vmrx
 
@@ -315,7 +315,7 @@ class ReadEEAAQEREPBase(ReadUngriddedBase):
             if idx in time_indexes:
                 data_dict[header[idx]] = np.zeros(self.MAX_LINES_TO_READ, dtype="datetime64[s]")
             else:
-                data_dict[header[idx]] = np.empty(self.MAX_LINES_TO_READ, dtype=np.float_)
+                data_dict[header[idx]] = np.empty(self.MAX_LINES_TO_READ, dtype=np.float64)
 
         # read the data...
         # DE,http://gdi.uba.de/arcgis/rest/services/inspire/DE.UBA.AQD,NET.DE_BB,STA.DE_DEBB054,DEBB054,SPO.DE_DEBB054_PM2_dataGroup1,SPP.DE_DEBB054_PM2_automatic_light-scat_Duration-30minute,SAM.DE_DEBB054_2,PM2.5,http://dd.eionet.europa.eu/vocabulary/aq/pollutant/6001,hour,3.2000000000,µg/m3,2020-01-04 00:00:00 +01:00,2020-01-04 01:00:00 +01:00,1,2
@@ -353,7 +353,7 @@ class ReadEEAAQEREPBase(ReadUngriddedBase):
                     # data is not a time
                     # sometimes there's no value in the file. Set that to nan
                     try:
-                        data_dict[header[idx]][lineidx] = np.float_(rows[idx])
+                        data_dict[header[idx]][lineidx] = np.float64(rows[idx])
                     except (ValueError, IndexError):
                         data_dict[header[idx]][lineidx] = np.nan
 

--- a/pyaerocom_env.yml
+++ b/pyaerocom_env.yml
@@ -3,7 +3,7 @@
 channels:
   - conda-forge
 dependencies:
-  - iris >=3.8.1
+  - iris >=3.11.0
   - xarray >=2022.12.0
   - cartopy >=0.21.1
   - matplotlib-base >=3.7.1
@@ -31,7 +31,7 @@ dependencies:
   - pip:
     - geojsoncontour
     - geocoder_reverse_natural_earth >= 0.0.2
-    - pyaro >= 0.0.12
+    - pyaro >= 0.0.14
     - aerovaldb >= 0.1.1
 ## testing
   - pytest >=7.4

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -47,7 +47,7 @@ dependencies = [
     'cf-units>=3.3.0',
     "pydantic>=2.7.1",
     "pyproj>=3.0.0",
-    "pyaro @ git+ssh://git@github.com/metno/pyaro@bump-numpy",
+    "pyaro @ git+https://git@github.com/metno/pyaro@bump-numpy",
     "pooch>=1.7.0",
     "psutil>=5.0.0",
 ]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,7 +22,7 @@ classifiers = [
 requires-python = ">=3.10"
 dependencies = [
     "aerovaldb>=0.1.1",
-    "scitools-iris==3.11.0rc0",
+    "scitools-iris>=3.11.0",
     "xarray>=2022.12.0",
     "cartopy>=0.21.1",
     "matplotlib>=3.7.1",
@@ -47,7 +47,7 @@ dependencies = [
     'cf-units>=3.3.0',
     "pydantic>=2.7.1",
     "pyproj>=3.0.0",
-    "pyaro @ git+https://git@github.com/metno/pyaro@bump-numpy",
+    "pyaro>=0.0.14",
     "pooch>=1.7.0",
     "psutil>=5.0.0",
 ]
@@ -215,7 +215,7 @@ extras =
 setenv =
     UDUNITS2_XML_PATH=/usr/share/xml/udunits/udunits2-common.xml
 deps =
-    scitools-iris ==3.11.0rc0; python_version < "3.11"
+    scitools-iris ==3.11.0; python_version < "3.11"
     cartopy ==0.21.1; python_version < "3.11"
     matplotlib ==3.7.1; python_version < "3.11"
     scipy ==1.10.1; python_version < "3.11"
@@ -228,7 +228,7 @@ deps =
     typing-extensions ==4.6.1; python_version < "3.11"
     cf-units ==3.3.0; python_version < "3.11"
     pydantic ==2.7.1; python_version < "3.11"
-    pyaro == 0.0.10; python_version < "3.11"
+    pyaro == 0.0.14; python_version < "3.11"
     pooch ==1.7.0; python_version < "3.11"
     xarray ==2022.12.0; python_version < "3.11"
     pandas ==1.5.3; python_version < "3.11"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,13 +22,13 @@ classifiers = [
 requires-python = ">=3.10"
 dependencies = [
     "aerovaldb>=0.1.1",
-    "scitools-iris>=3.8.1",
+    "scitools-iris==3.11.0rc0",
     "xarray>=2022.12.0",
     "cartopy>=0.21.1",
     "matplotlib>=3.7.1",
     "scipy>=1.10.1",
     "pandas>=1.5.3",
-    "numpy>=1.24.4, <2.0.0",
+    "numpy>=1.24.4",
     "seaborn>=0.12.2",
     "dask",
     "geonum==1.5.0",
@@ -44,10 +44,10 @@ dependencies = [
     'importlib-resources>=5.10; python_version < "3.11"',
     'typing-extensions>=4.6.1; python_version < "3.11"',
     # https://github.com/SciTools/cf-units/issues/218
-    'cf-units>=3.1',
+    'cf-units>=3.3.0',
     "pydantic>=2.7.1",
     "pyproj>=3.0.0",
-    "pyaro>=0.0.12",
+    "pyaro @ git+ssh://git@github.com/metno/pyaro@bump-numpy",
     "pooch>=1.7.0",
     "psutil>=5.0.0",
 ]
@@ -215,7 +215,7 @@ extras =
 setenv =
     UDUNITS2_XML_PATH=/usr/share/xml/udunits/udunits2-common.xml
 deps =
-    scitools-iris ==3.8.1; python_version < "3.11"
+    scitools-iris ==3.11.0rc0; python_version < "3.11"
     cartopy ==0.21.1; python_version < "3.11"
     matplotlib ==3.7.1; python_version < "3.11"
     scipy ==1.10.1; python_version < "3.11"
@@ -226,7 +226,7 @@ deps =
     tomli ==2.0.1; python_version < "3.11"
     importlib-resources ==5.10; python_version < "3.11"
     typing-extensions ==4.6.1; python_version < "3.11"
-    cf-units ==3.1; python_version < "3.11"
+    cf-units ==3.3.0; python_version < "3.11"
     pydantic ==2.7.1; python_version < "3.11"
     pyaro == 0.0.10; python_version < "3.11"
     pooch ==1.7.0; python_version < "3.11"

--- a/scripts/aeolus2netcdf.py
+++ b/scripts/aeolus2netcdf.py
@@ -60,10 +60,10 @@ def main():
         help="set path of CODA_DEFINITION env variable",
         default="/lustre/storeA/project/aerocom/aerocom1/ADM_CALIPSO_TEST/",
     )
-    parser.add_argument("--latmin", help="min latitude to return", default=np.float_(30.0))
-    parser.add_argument("--latmax", help="max latitude to return", default=np.float_(76.0))
-    parser.add_argument("--lonmin", help="min longitude to return", default=np.float_(-30.0))
-    parser.add_argument("--lonmax", help="max longitude to return", default=np.float_(45.0))
+    parser.add_argument("--latmin", help="min latitude to return", default=np.float64(30.0))
+    parser.add_argument("--latmax", help="max latitude to return", default=np.float64(76.0))
+    parser.add_argument("--lonmin", help="min longitude to return", default=np.float64(-30.0))
+    parser.add_argument("--lonmax", help="max longitude to return", default=np.float64(45.0))
     parser.add_argument(
         "--dir",
         help="work on all files below this directory",
@@ -186,16 +186,16 @@ def main():
         options["tempdir"] = args.tempdir
 
     if args.latmin:
-        options["latmin"] = np.float_(args.latmin)
+        options["latmin"] = np.float64(args.latmin)
 
     if args.latmax:
-        options["latmax"] = np.float_(args.latmax)
+        options["latmax"] = np.float64(args.latmax)
 
     if args.lonmin:
-        options["lonmin"] = np.float_(args.lonmin)
+        options["lonmin"] = np.float64(args.lonmin)
 
     if args.lonmax:
-        options["lonmax"] = np.float_(args.lonmax)
+        options["lonmax"] = np.float64(args.lonmax)
 
     if args.emep:
         options["emepflag"] = args.emep
@@ -437,7 +437,7 @@ def main():
                 nc_longitudes = nc_data["lon"].data
                 nc_lev_no = len(nc_data["lev"])
                 nc_colocated_data = np.zeros(
-                    [aeolus_profile_no * nc_lev_no, obj._COLNO], dtype=np.float_
+                    [aeolus_profile_no * nc_lev_no, obj._COLNO], dtype=np.float64
                 )
 
             # locate current rounded Aeolus time in netcdf file

--- a/scripts/read_sentinel5p.py
+++ b/scripts/read_sentinel5p.py
@@ -52,10 +52,10 @@ def main():
     )
     # parser.add_argument("--codadef", help="set path of CODA_DEFINITION env variable",
     #                     default='/lustre/storeA/project/aerocom/aerocom1/ADM_CALIPSO_TEST/')
-    parser.add_argument("--latmin", help="min latitude to return", default=np.float_(30.0))
-    parser.add_argument("--latmax", help="max latitude to return", default=np.float_(76.0))
-    parser.add_argument("--lonmin", help="min longitude to return", default=np.float_(-30.0))
-    parser.add_argument("--lonmax", help="max longitude to return", default=np.float_(45.0))
+    parser.add_argument("--latmin", help="min latitude to return", default=np.float64(30.0))
+    parser.add_argument("--latmax", help="max latitude to return", default=np.float64(76.0))
+    parser.add_argument("--lonmin", help="min longitude to return", default=np.float64(-30.0))
+    parser.add_argument("--lonmax", help="max longitude to return", default=np.float64(45.0))
     parser.add_argument(
         "--dir",
         help="work on all files below this directory",
@@ -152,16 +152,16 @@ def main():
         options["tempdir"] = args.tempdir
 
     if args.latmin:
-        options["latmin"] = np.float_(args.latmin)
+        options["latmin"] = np.float64(args.latmin)
 
     if args.latmax:
-        options["latmax"] = np.float_(args.latmax)
+        options["latmax"] = np.float64(args.latmax)
 
     if args.lonmin:
-        options["lonmin"] = np.float_(args.lonmin)
+        options["lonmin"] = np.float64(args.lonmin)
 
     if args.lonmax:
-        options["lonmax"] = np.float_(args.lonmax)
+        options["lonmax"] = np.float64(args.lonmax)
 
     if args.emep:
         options["emepflag"] = args.emep

--- a/tests/aeroval/test_json_utils.py
+++ b/tests/aeroval/test_json_utils.py
@@ -29,13 +29,13 @@ def json_path(tmp_path: Path) -> Path:
             id="single float",
         ),
         pytest.param(
-            [np.float_(2.3456789), np.float32(3.456789012)],
+            [np.float64(2.3456789), np.float32(3.456789012)],
             3,
             [2.346, pytest.approx(3.457, 1e-3)],
             id="np.float list",
         ),
         pytest.param(
-            (np.float128(4.567890123), np.float_(5.6789012345)),
+            (np.float128(4.567890123), np.float64(5.6789012345)),
             5,
             [pytest.approx(4.56789, 1e-5), 5.67890],
             id="np.float tuple",

--- a/tests/fixtures/collocated_data.py
+++ b/tests/fixtures/collocated_data.py
@@ -246,7 +246,7 @@ def _create_fake_partial_trends_coldata_3d(colocate_time: bool = False):
             if d in dtimes[i]:
                 data[0, j, i] = 1
             else:
-                data[0, j, i] = np.NaN
+                data[0, j, i] = np.nan
 
     meta = {
         "data_source": ["fakeobs", "fakemod"],

